management_api_app/db/repositories/workspace_templates.py

```diff
@@ -0,0 +1,43 @@
+from typing import List
+
+from azure.cosmos import CosmosClient
+
+from core import config
+from db.errors import EntityDoesNotExist
+from db.repositories.base import BaseRepository
+from models.domain.resource_template import ResourceTemplate
+
+
+class WorkspaceTemplateRepository(BaseRepository):
+    def __init__(self, client: CosmosClient):
+        super().__init__(client, config.STATE_STORE_RESOURCE_TEMPLATES_CONTAINER)
+
+    @staticmethod
+    def _workspace_template_by_name_query(name: str) -> str:
+        return f'SELECT * FROM c WHERE c.resourceType = "workspace" AND c.name = "{name}"'
+
+    def get_workspace_templates_by_name(self, name: str) -> List[ResourceTemplate]:
+        query = self._workspace_template_by_name_query(name)
+        return self._query(query=query)
+
+    def get_current_workspace_template_by_name(self, name: str) -> dict:
+        query = self._workspace_template_by_name_query(name) + ' AND c.isCurrent = true'
+        workspace_templates = self._query(query=query)
+        if len(workspace_templates) != 1:
+            raise EntityDoesNotExist
+        return workspace_templates[0]
+
+    def get_workspace_template_by_name_and_version(self, name: str, version: str) -> dict:
+        query = self._workspace_template_by_name_query(name) + f' AND c.version = "{version}"'
+        workspace_templates = self._query(query=query)
+        print(len(workspace_templates))
+        print(workspace_templates)
+        if len(workspace_templates) != 1:
+            raise EntityDoesNotExist
+        return workspace_templates[0]
+
+    def get_workspace_template_names(self) -> List[str]:
+        query = 'SELECT * FROM c'
```

Comment:
why not database handle the projection??
SELECT {"Name":f.name} FROM WorkspaceTemplates f 
isn't that better we only get the stuff we need from the DB instead of loading the whole content in memory?